drivers/serial/leuart_gecko.c

```diff
@@ -0,0 +1,385 @@
+
+	/* The peripheral and gpio clock are already enabled from soc and gpio
+	 * driver
+	 */
+
+	leuartInit.baudrate = config->baud_rate;
+
+	/* Enable LEUART clock */
+	CMU_ClockEnable(config->clock, true);
+
+	/* Enable CORE LE clock in order to access LE modules */
+	CMU_ClockEnable(cmuClock_CORELE, true);
+
+	/* Select LFXO for LEUARTs (and wait for it to stabilize) */
+	CMU_ClockSelectSet(cmuClock_LFB, cmuSelect_LFXO);
+
+	/* Init LEUART */
+	LEUART_Init(base, &leuartInit);
+
+	/* Initialize LEUART pins */
+	leuart_gecko_init_pins(dev);
+
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	config->irq_config_func(dev);
+#endif
+
+	return 0;
+}
+
+static const struct uart_driver_api leuart_gecko_driver_api = {
+	.poll_in = leuart_gecko_poll_in,
+	.poll_out = leuart_gecko_poll_out,
+	.err_check = leuart_gecko_err_check,
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	.fifo_fill = leuart_gecko_fifo_fill,
+	.fifo_read = leuart_gecko_fifo_read,
+	.irq_tx_enable = leuart_gecko_irq_tx_enable,
+	.irq_tx_disable = leuart_gecko_irq_tx_disable,
+	.irq_tx_complete = leuart_gecko_irq_tx_complete,
+	.irq_tx_ready = leuart_gecko_irq_tx_ready,
+	.irq_rx_enable = leuart_gecko_irq_rx_enable,
+	.irq_rx_disable = leuart_gecko_irq_rx_disable,
+	.irq_rx_ready = leuart_gecko_irq_rx_ready,
+	.irq_err_enable = leuart_gecko_irq_err_enable,
+	.irq_err_disable = leuart_gecko_irq_err_disable,
+	.irq_is_pending = leuart_gecko_irq_is_pending,
+	.irq_update = leuart_gecko_irq_update,
+	.irq_callback_set = leuart_gecko_irq_callback_set,
+#endif
+};
+
+#ifdef CONFIG_LEUART_GECKO_0
+
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+static void leuart_gecko_config_func_0(struct device *dev);
+#endif
+
+static const struct leuart_gecko_config leuart_gecko_0_config = {
+	.base = (LEUART_TypeDef *) LEUART_0_BASE_ADDRESS,
+	.clock = cmuClock_LEUART0,
+	.baud_rate = LEUART_0_CURRENT_SPEED,
+	.pin_rx = PIN_LEUART0_RXD,
+	.pin_tx = PIN_LEUART0_TXD,
+	.loc = LEUART_0_LOCATION,
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	.irq_config_func = leuart_gecko_config_func_0,
+#endif
+};
+
+static struct leuart_gecko_data leuart_gecko_0_data;
+
+DEVICE_AND_API_INIT(leuart_0, LEUART_0_LABEL, &leuart_gecko_init,
+		    &leuart_gecko_0_data, &leuart_gecko_0_config, PRE_KERNEL_1,
+		    CONFIG_KERNEL_INIT_PRIORITY_DEVICE,
+		    &leuart_gecko_driver_api);
+
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+static void leuart_gecko_config_func_0(struct device *dev)
+{
+	IRQ_CONNECT(LEUART_0_IRQ, LEUART_0_IRQ_PRIORITY, leuart_gecko_isr,
+		    DEVICE_GET(leuart_0), 0);
+
+	irq_enable(LEUART_0_IRQ);
+}
+#endif
+
+#endif /* CONFIG_LEUART_GECKO_0 */
+
+#ifdef CONFIG_LEUART_GECKO_1
+
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+static void leuart_gecko_config_func_1(struct device *dev);
+#endif
+
+static const struct leuart_gecko_config leuart_gecko_1_config = {
+	.base = (LEUART_TypeDef *) LEUART_1_BASE_ADDRESS,
+	.clock = cmuClock_LEUART0,
+	.baud_rate = LEUART_1_CURRENT_SPEED,
+	.pin_rx = PIN_LEUART0_RXD,
```

Comment:
Should PIN_LEUART0_RXD be PIN_LEUART1_RXD (also in the next line)?

Reply:
Fixed